student-info-system/sis/models.py

```diff
@@ -576,27 +594,79 @@ def seats_remaining(self):
     def __str__(self):
         return self.name
 
+    # the prof may have updated them; we may have changed professors...whyever
+    def refresh_reference_items(self):
+        for item in self.sectionreferenceitem_set.all():
+            item.delete()
+        if self.professor is not None:
+            ix = 1
+            for item in self.professor.referenceitem_set.filter(course=self.course):
+                SectionReferenceItem.objects.create(item=item, section=self, index=ix)
+                ix = ix + 1
+
+
+class ReferenceItem(models.Model):
```

Comment:
True, more so in this case then the others because this was a specifc new feature that didn't break much (if anything). I think I did find profile issues while doing this IIRC. Will review the commit again...